qt-models/divelocationmodel.cpp

```diff
@@ -183,6 +169,56 @@ void LocationInformationModel::diveSiteDeleted(struct dive_site *, int idx)
 	endRemoveRows();
 }
 
+bool DiveSiteSortedModel::filterAcceptsRow(int sourceRow, const QModelIndex &source_parent) const
+{
+	// TODO: filtering
+	return true;
+}
+
+bool DiveSiteSortedModel::lessThan(const QModelIndex &i1, const QModelIndex &i2) const
+{
+	// The source indices correspond to indices in the global dive site table.
+	// Let's access them directly without going via the source model.
+	// Kind of dirty, but less effort.
+	struct dive_site *ds1 = get_dive_site(i1.row(), &dive_site_table);
+	struct dive_site *ds2 = get_dive_site(i2.row(), &dive_site_table);
+	if (!ds1 || !ds2) // Invalid dive sites compare as different
+		return false;
+	switch (i1.column()) {
+	case LocationInformationModel::NAME:
+	default:
+		return QString::localeAwareCompare(QString(ds1->name), QString(ds2->name)) < 0; // TODO: avoid copy
+	case LocationInformationModel::DESCRIPTION: {
+		int cmp = QString::localeAwareCompare(QString(ds1->description), QString(ds2->description)); // TODO: avoid copy
+		return cmp != 0 ? cmp < 0 :
+		       QString::localeAwareCompare(QString(ds1->name), QString(ds2->name)) < 0; // TODO: avoid copy
+	}
+	case LocationInformationModel::NUM_DIVES: {
+		int cmp = ds1->dives.nr - ds2->dives.nr;
+		// Since by default nr dives is descending, invert sort direction of names, such that
+		// the names are listed as ascending.
+		return cmp != 0 ? cmp < 0 :
+		       QString::localeAwareCompare(QString(ds1->name), QString(ds2->name)) < 0; // TODO: avoid copy
+	}
+	}
+}
+
+DiveSiteSortedModel::DiveSiteSortedModel()
+{
+	setSourceModel(LocationInformationModel::instance());
```

Comment:
so this sets the `DiveSiteSortedModel` - but I don't see how this would propagate into the `TabDiveSite::model`

Reply:
The details are implemented by Qt's `QSortFilterProxyModel` class. You just have to set the source model and that's it. Out filter-code could definitely be simplified.